File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
  const clothesMap = {};
  for (let i = 0; i < clothes.length; i++) {
    let type = clothes[i][1];

    if (!clothesMap[type]) {
      clothesMap[type] = 1;
    } else {
      clothesMap[type]++;
    }
  }

  let answer = 1;
  for (let type in clothesMap) {
    answer *= clothesMap[type] + 1;
  }

  return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    hash = {} 
    for item in clothes:
        if item[1] in hash:
            hash[item[1]].append(item[0]) 
        else: 
            hash[item[1]] = [item[0]]
            
    combination = 1
    for key in hash:
        combination *= len(hash[key]) + 1
        
    return (combination - 1)
        
        